### DISCUSSION TITLE
🆕 Product roadmap for GitHub Issues

### DISCUSSION THREAD

**Discussion by azenMatt** [community/community#6803] [Projects and Issues]
👋  Hi everyone, we recently published a new [public roadmap](https://github.com/orgs/github/projects/4247) using projects beta, which has an [Issues-specific view](https://github.com/orgs/github/projects/4247/views/7). So you can now see all the planning and tracking related work we’re doing in one place! 

We'd love your feedback on both the value of having separate views in this roadmap for each category and the contents of the Issues view. 

- Are we prioritising the right things in the near team? 
- Do you agree with what is coming in the future - are we missing anything you'd love to see us thinking about? 

Right now the contents of these issues are a little empty, but I'll be updating before the end of the year to have more content and design ideas to help bring them to life for you all a little.

**Comment by dodyg** [comment]
Oh I was expecting the Timelines layout to be ready sooner than "Future" because the marketing page says it's "coming soon". 

![FCmp4bTWEAMKsk2](https://user-images.githubusercontent.com/1761249/139050840-c4a3bd90-de84-4268-b567-970cacee64eb.jpg)

**Reply by galah92** [reply]
Expecting this as well.

**Reply by erikr** [reply]
Can't wait for this!

**Reply by alexcnichols** [reply]
👋🏼 Improvements for views, including a new timeline view, are on the roadmap. Check out this roadmap item for more on timeline and subscribe to be notified if/when things change. 

- https://github.com/github/roadmap/issues/278

**Reply by ant-04** [reply]
We need the timeline view sooner 😀

**Reply by erikr** [reply]
@alexcnichols Can you delete the spam posts from @gazegAonad here? Thanks.

**Comment by jesfer** [comment]
Separate views is awesome. Keep it.

I agree with everything on the list as well as the order of the items. Only problem I see is you cant deliver this fast enough as I want it all yesterday 😉

**Comment by DmitriiKozhevnikov** [comment]
Hey! Super glad to hear you intend putting more effort into issues and projects since we use this feature extensively in our company.

Adding new projects to repo-level is definitely the right move. My favourites from the roadmap would be (ordered by my personal preference):

1. Custom automation flows - supporting workflows for issue automation is somewhat bothersome, and rolling out ubuntu vm each time I need to move a card just irks me.
2. Custom metadata - using labels for everything creates a colourful mess on project board.
3. Hierarchy navigation / projects in projects - ngl don't see how this is different, but having different levels of detail for issues would surely come in handy.

Other than that I mostly miss some basic kanban stuff:

- Explicit blockers. Ability to state the reason for blocking would also be nice. Currently I use a workaround good ol' red "Blocked" label and a comment explaining why this issue is blocked.
- Work in progress limits. This is one of those things that keeps your kanban running. Simply painting the project column red when the number of cards in it exceeds some number would be very useful.
- Project analytics. Right now I have to dump issue history and analyze it manually or via scripts which is just tiresome. Built-in charts like cumulative flow diagram, or cycle time histogram would help a lot.

**Reply by RickCogley** [reply]
agreed. Without WIP limits and some way to clearly indicate in the state column what's the "definition of done", it's not really even kanban. The former can be solved simply as @DmitriiKozhevnikov mentions, being able to set a WIP limit (number of cards in the column) and changing the color to alert people. The latter with a note area above or below the column, to show "what this is; what does done mean".

**Reply by kikomaxx22** [reply]
Huh I’m lost already

**Reply by erikr** [reply]
These are good ideas.

**Comment by sayyid5416** [comment]
[Auto add issue from repo to project](https://github.com/github/roadmap/issues/286)  is really important, I hope it comes sooner than 'future'.

**Reply by redcatbear** [reply]
I absolutely agree. While I found a workaround with the [project-bot](https://github.com/philschatz/project-bot) app, that one only works with the "old" projects, since it requires a note with the workflow rules on the board and the new boards don't support notes (yet).

Auto-adding issues to boards is absolutely essential. There are so many things that can go wrong if you do it by hand, like forgetting it or moving to the wrong board.

**Comment by Bejuna055** [comment]
I want add issue repo to project on my account
I need to spon and feedback or transfer on new account or my email

**Comment by slinna** [comment]
I definitely like the feature set you're planning to build for the Projects (beta)! However, I didn't see anything related to the API support. The new current Projects (beta) API is missing some features that the current Projects API supports. 

For example, activities within the Project don't create any timeline items inside an Issue. If I change the name of an Issue, in legacy projects API, I have a new timeline item that would show that change. Or in case if I add some Issue or PR inside a legacy project then I will have an event for that. In the new API, these are all missing.

Also, it would be nice to get all the connected projects to my issue via the API.

I would love to get some clarity on when these things might be supported in the new API.

**Reply by gazegAonad** [reply]
helo sir

**Comment by sayyid5416** [comment]
@slinna `activities within the Project don't create any timeline`

I agree with you on this.

**Reply by gazegAonad** [reply]
ariba

**Comment by gazegAonad** [comment]
ariab
![12](https://user-images.githubusercontent.com/93989075/141107578-89319463-b4ef-4df4-8de4-79d2da38f160.jpg)

**Comment by AninaL** [comment]
would love to see the following stuff in the future:
- creating sprints in github (using at the moment milestones as a workaround)
- linking dokuments and pages for the team
- insights for productowner
- we love the diffrent views, but would love to have some improvement on the sorting like sorting in groups
- would be also nice to create issues in the project for a specific repo

**Reply by gazegAonad** [reply]
hi

**Reply by drakored** [reply]
Have you considered using Azure Devops for the Sprint board/scrum features and using the integration between them for tagging PR and commits to stories?  The integration between them isn't quite as cool as all in one platform, but ADO has a more mature set of Scrum planning features, and with some work you can blend them decently.

**Reply by Deborah-Digges** [reply]
> creating sprints in github (using at the moment milestones as a workaround)

Have you considered using the `Iteration` field type to support sprints? It's a new field type we just introduced in October, and you can find more information about it here: https://github.blog/changelog/2021-10-14-the-new-github-issues-10-14-update/

**Comment by futuredesignUK** [comment]
@azenMatt Great demo of the new issues at Universe and good to see it in public beta.

It's getting closer to Azure Boards now, so maybe time to switch from ADO.

The public roadmap looks nice. Couple of things in the wish list:

- Issues estimation points / hours 
- From project view can we get a modal/pop up of the issue before the open new window to issue
- Sprint burndown & reporting
- Filter on 'Current Sprint' instead of manually changing Sprint 2/3 etc

![image](https://user-images.githubusercontent.com/19208973/141119046-65876455-42ee-43c7-99e8-d35d3bcf42a6.png)



I've put up a blog post on this as it's a great topic - nice work

https://sam.hope-evans.com/github-issues-vs-azure-boards/

**Reply by AninaL** [reply]
i see you wish to have the same things like me. 

>Filter on 'Current Sprint' instead of manually changing Sprint 2/3 etc

for this i found the solution to use Milstones as sprints and then just having Sprint Backlog, Sprint and Sprint done as Milestones. Issues who are done will be put into the Sprint done milestone. If you would like to have a look at the old sprints, then of course you can use labels with sprint 1, 2....

**Reply by Littlewillyjr** [reply]
Hello aninal can we work on a new skill.

**Reply by AninaL** [reply]
btw: even better to create an additional field with iteration.

**Reply by azenMatt** [reply]
Thanks @futuredesignUK - all those changes are on our radar :) 

For 
>Issues estimation points / hours

Is there something in particular missing from using a number field for this at the moment?

**Reply by futuredesignUK** [reply]
@azenMatt 
> Thanks @futuredesignUK - all those changes are on our radar :)
> 
> For
> 
> > Issues estimation points / hours
> 
> Is there something in particular missing from using a number field for this at the moment?

No you are right - this can already be done! In fact even better just knocked up animal size for effort

![image](https://user-images.githubusercontent.com/19208973/141489223-0378e819-6717-4fd9-8a53-3f479c7a9b3b.png)

**Reply by vojtech-dobes** [reply]
> Thanks @futuredesignUK - all those changes are on our radar :)
> 
> For
> 
> > Issues estimation points / hours
> 
> Is there something in particular missing from using a number field for this at the moment?

The number fields are awesome - but is it somehow possible to get sum of the values for given column / group selection etc.? If I haven't missed something, they work only as standalone values, there is no kind of aggregation available.

**Reply by futuredesignUK** [reply]
> 

@vojtech-dobes 

Yes that would be great for sprint planning etc  - I suspect it might come under the Project Insights issue

https://github.com/github/roadmap/issues/279

**Comment by mt-development** [comment]
Would be great to see that earlier than future :)
- [ ] https://github.com/github/roadmap/issues/286
For new Projects (beta).

**Reply by andrewking1235** [reply]
Agreed 💯. This would be useful in the case where you want to keep a high-level overview of all the issues, but don't always enter them yourself (or sometimes forget to add an issue to your project). If you or anyone else do forget, it's really tedious to go through all the issues one-by-one and check to make sure they're in the project. But the alternative is not knowing if you can trust that you're really seeing everything. Thank you!!

**Comment by webern** [comment]
I'm looking for support of "sprints" or "iterations" but not finding those keywords. Is it anticipated that projects will support sprint workflows?
- Creating sprints
- Adding items to a sprint
- Tracking sprint progress

**Reply by azenMatt** [reply]
Have you tried out the iterations field type?

**Comment by cilerler** [comment]
I would like to be able to create columns from the labels. like `=labelprefix('Effort: ")` where labels may look like 

```txt
| Label                             | Color   |
| --------------------------------- | ------- |
| Effort: 2                         | #E1BEE7 |
| Effort: 4                         | #E1BEE7 |
| Effort: 8                         | #E1BEE7 |
| Effort: 16                        | #E1BEE7 |
| Effort: 24                        | #E1BEE7 |
| Effort: 32                        | #E1BEE7 |
| Effort: 40                        | #E1BEE7 |
| Effort: 80                        | #E1BEE7 |
| Effort: 120                       | #E1BEE7 |
| Effort: 160                       | #E1BEE7 |
| Effort: TBD                       | #7B1FA2 |
```
Click [here](https://github.com/cilerler/cilerler.github.io/wiki/Labels) for the complete label list.

![image](https://user-images.githubusercontent.com/385958/141969905-8456c218-7421-4adf-b231-819ae716d548.png)

**Comment by jasononeil** [comment]
For https://github.com/github/roadmap/issues/283

One of the biggest pet peeves for me and my team at the moment is we filter our Kanban board to the current "iteration". When we try to create a new card on that board... it disappears, because there is no current iteration set.

As far as I can tell there is no way to create a new card/row on a filtered view, and for it to stay visible long enough to set the fields that would make it appear on the filtered view.

Possible solutions:
- A new card/row on a filtered view automatically contains the field values required to be visible on that view
- A new card/row stays visible after creating it, long enough for you to manually set the fields so it will appear in the current list

**Reply by azenMatt** [reply]
Hold tight - this is coming next week 😄 

We are going to apply the filtered values to the card (where possible)

**Reply by jasononeil** [reply]
That and the lack of “bulk updates” where my two biggest pet peeves and it sounds like both will be resolved soon. Can’t wait!

**Reply by futuredesignUK** [reply]
Good point @jasononeil 

I was trying this earlier - great to see it being sorted

![image](https://user-images.githubusercontent.com/19208973/143677572-9f9ae227-2cb1-4963-8a86-901fe9134230.png)

One other thing @azenMatt  - feature to clone project settings (custom fields and views etc) saves recreating them all again in a new project. In essence project templates

**Reply by azenMatt** [reply]
We are just putting the finishing touches to those filter updates - we've been testing it internally and for a small tweak it makes life so much better! Will be with you soon. 

👍  on templates - that is exactly where we want to go with the product in the future 😄  🚀

**Reply by azenMatt** [reply]
This is now live! check it out here - https://github.blog/changelog/2021-12-02-the-new-github-issues-12-02-update/

**Comment by hbulgarini** [comment]
Nice work on all the new features being created!

it would be great to have the chance to establish dependencies between issues and have the chance to view them in a timeline with their corresponding relations.

That would be a game changer for sure.

**Comment by liamsi** [comment]
I just realized that one can't associate a beta project with a team currently:
![image](https://user-images.githubusercontent.com/7441766/144465282-bc4cd797-28bd-40a8-94f8-af67f6b68b53.png)

--------


Only the "old"/regular projects can be found. Any timeline for this?

**Comment by cilerler** [comment]
We need (1) cross-repository milestones and (2) be able to group by based on parent issues that are set in the task list.

![image](https://user-images.githubusercontent.com/385958/144725202-ec9231ae-f890-4fad-bc6b-812563a0ff12.png)

**Reply by sayyid5416** [reply]
Cross repo milestones would be very useful

**Comment by cynthiabellmcgillis** [comment]
Am I an idiot or is there no way to bulk add issues to a project?

**Reply by macMikey** [reply]
Here's how I've done it:
* Make sure you have the workflow set to auto-add new issues to the project
* In the repo, assign the project (menu next to "issues")
* Go to Issues, do a select all, then choose the project. You have to do this for each page of issues, but then all the issues in the repo will go into the project.

**Comment by Rovel** [comment]
Hi I just started using Project Beta and is amazing, I know its been in beta for a while, so any schedule on:
- When Projects Insights will be on beta, currently its on alpha, so any chance promoting this year? (BTW, I really miss insights data on Github in general)
- Any chance of the projects beta can add all repo issues? currently I need to add one by one right?

**Reply by ElijahLynn** [reply]
It is still in the "Status: Future" state per https://github.com/orgs/github/projects/4247/views/7.

![image](https://user-images.githubusercontent.com/1504756/166300403-6b327052-f16a-47bc-9f2f-fcb85fdaaff1.png)


[Automatically add new issues from a repository into projects #286](https://github.com/github/roadmap/issues/286) (consider hitting the "subscribe" button on this issue if you want to be notified upon new activity)

**Comment by macMikey** [comment]
mobile: add projects, etc.

**Reply by macMikey** [reply]
Mobile anything at this point would be great
